pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -523,25 +523,31 @@ private CompletableFuture<Long> individualAckWithTransaction(CommandAck ack) {
         LongAdder totalAckCount = new LongAdder();
         for (int i = 0; i < ack.getMessageIdsCount(); i++) {
             MessageIdData msgId = ack.getMessageIdAt(i);
-            PositionImpl position;
+            PositionImpl position = PositionImpl.get(msgId.getLedgerId(), msgId.getEntryId());
+            // acked count at least one
             long ackedCount = 0;
-            long batchSize = getBatchSize(msgId);
+            long batchSize = 0;
```

Comment:
The default value may be 1.

Reply:
if not batch message, the batchSize can not be able to 1

Reply:
Ok, If not batch message, the batch size should be 0, right?